dpytools/http/upload.py

```diff
@@ -1,70 +1,169 @@
+        assert (
+            self.florence_user is not None
+        ), "Where env var SERVICE_TOKEN_FOR_UPLOAD is None, env var FLORENCE_PASSOWRD must be provided"
+        assert (
+            self.florence_user is not None
+        ), "Where env var SERVICE_TOKEN_FOR_UPLOAD is None, env var IDENTITY_API_URL must be provided"
+
+        # https://github.com/ONSdigital/dp-identity-api/blob/develop/swagger.yaml
+        token_url = f"{self.identity_api_url}/tokens"
+        response = self.post(
+            token_url,
+            json={"login": self.florence_user, "password": self.florence_password},
+        )
+        if response.status_code == 201:
+            response_headers = response.headers
+
+            self.refresh_token = response_headers["Refresh"]
+            self.token_creation_time = datetime.now()
+
+            self.auth_token = response_headers["Authorization"]
+            self.id_token = response_headers["ID"]
+        else:
+            err = Exception("Failed to create user tokens")
+            logger.error(
+                "Failed to create user tokens",
+                err,
+                data={
+                    "identity_api_url": self.identity_api_url,
+                    "token_url": token_url,
+                    "response_staus_code": response.status_code,
+                    "response_content": response.content,
+                },
+            )
+            raise err
+
+    def get_auth_header(self) -> Dict[str, str]:
+        """
+        Given a dictionary of params, set the auth header based on the
+        auth mode in use.
+        """
+
+        # Using service account
+        if self.service_token:
+            return {"Authorization": f"Bearer {self.service_token}"}
+
+        # Using user account
+        # If the token is more than 10 minutes old refresh it
+        # https://github.com/ONSdigital/dp-identity-api/blob/develop/swagger.yaml
+        if (datetime.now() - self.token_creation_time) > timedelta(minutes=10):
+            token_refresh_url = f"{self.identity_api_url}/tokens/self"
+            response = self.put(
+                token_refresh_url,
+                json={"Refresh": self.refresh_token, "ID": self.id_token},
+            )
+
+        if response.status_code == 201:
+            self.auth_token = response.headers["Authorization"]
+            self.id_token = response.headers["ID"]
+        else:
+            err = Exception(
+                f"Refreshing token failed, returned a {response.status_code} error"
+            )
+            logger.error(
+                "Could not refresh user auth token",
+                err,
+                data={
+                    "token_refresh_url": token_refresh_url,
+                    "response_status_code": response.status_code,
+                    "response_content": response.content,
+                },
+            )
+            raise err
```

Comment:
This should be indented further shouldn't it?

Reply:
I dont think so on this one.

- we create but dont raise an error on 117.
- we do some logging things
- we raise that error from 117.

does that make sense or am I missing a trick?

Reply:
I think it needs indenting - when I tried to run it without indenting, I got a `UnboundLocalError: cannot access local variable 'response' where it is not associated with a value`

Reply:
....yup you're both right and Mike should have had that coffee. gotta have a response to look at a response